### ISSUE TITLE
edge_inserts.dart

### ISSUE THREAD

**Issue by patrikheinonen** [flutter/flutter#115951]
Some things found about the classes in this file.

-In edge_insets.dart on lines 894 and 997 the assert(direction != null); is called even tho on lines 893 and 996 we accept a nullable parameter. 

-The constructor EdgeInsetsDirectional.all is more expensive than EdgeInsets to resolve and this hasn't been documented. Also it's fairly obvious that EdgeInsetsDirectional.all is redundant. 

-The way inheritance is used here could be argued as bad practice. Every sibling class(whose parent is EdgeInsetsGeometry) has to be aware of the EdgeInsets class.

-Someone experienced should review the whole file imo

**Comment by exaby73** [comment]
Hello @patrikheinonen. Could you include permalinks to the code in question for easier reference? You could also include the changes you would expect or better yet, file a PR if you want :)

**Comment by patrikheinonen** [comment]
Yeah @exaby73 Im very new to this open source stuff. This is actually the first issue I have ever raised so I didn't know about permalinks.
I just tried to point someones attention to the file. I will read the contributor rules etc and try to make a PR 👍 
what comes to the third point in my initial comment. The refactoring should prob be discussed about before deciding actions.

**Comment by exaby73** [comment]
Thanks for the information :) I'll label this issue for further discussion from the team. Thank you
